ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/BlockHeader.java

```diff
@@ -242,6 +254,7 @@ public static BlockHeader readFrom(
     final Hash mixHash = Hash.wrap(input.readBytes32());
     final long nonce = input.readLong();
     final Long baseFee = !input.isEndOfCurrentList() ? input.readLongScalar() : null;
+    final Bytes32 random = !input.isEndOfCurrentList() ? input.readBytes32() : null;
```

Comment:
Maybe I missed something but there is a mismatch between readFrom and writeTo.
- In WriteTo : we write Random or MixHash + we write it before the nonce
- In ReadTo : MixHash is not optional and we read random after the nonce and the baseFee

Reply:
Good catch.  Prior to amphora we (besu) were just appending random at the end, similar to baseFee.  EIP-4399 now says we need to reuse mixhash in order to save 32 bytes.  It isn't clear to me yet how we can differentiate between random and mixhash when we read blocks off the wire or from the db.

This implementation isn't ever going to read random, so we might as well remove the line until we fully address EIP-4399 in #3081

Reply:
I think we can repurpose mixHash as a generic Bytes32, and get rid of the explicit `random` constructor parameter.  Both random and mixhash would use the same storage and we differentiate between the two via the accessors.  It is a bit hacky imo, but the spec doesn't leave any other options I can see.